src/databricks/labs/ucx/source_code/queries.py

```diff
@@ -42,17 +49,22 @@ def lint(self, code: str) -> Iterable[Advice]:
                     end_col=1024,
                 )
 
-    @staticmethod
-    def _catalog(table):
+    def _catalog(self, table):
         if table.catalog:
             return table.catalog
-        return 'hive_metastore'
+        return self._use_schema if self._use_schema else 'hive_metastore'
 
     def apply(self, code: str) -> str:
         new_statements = []
         for statement in sqlglot.parse(code, read='databricks'):
             if not statement:
                 continue
+            if isinstance(statement, Use):
```

Comment:
Do we need this?

Reply:
I think so. I need to know that the statement is a USE statement to switch the schema for subsequent statements and subsequent linters.